### PR TITLE
Rework project repository clone

### PR DESCRIPTION
**Pull request by cutwater**
* Fix `git clone` waiting for interactive input when trying to clone private or non-existing GitHub repository.
* Add timeout for `git` commands.
* The `git_hash` project field is now read only.

Resolves: https://issues.redhat.com/browse/AAP-4833
Resolves: https://issues.redhat.com/browse/AAP-6185

### CODE REVIEW

**Comment by cutwater** [comment]
**Note:** Further refactoring is needed and will be done in subsequent PRs.

File: src/eda_server/utils/subprocess.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+TimeoutExpired = subprocess.TimeoutExpired
+
+CalledProcessError = subprocess.CalledProcessError
+
+CompletedProcess = subprocess.CompletedProcess
+
+
+async def run(
+    *cmd: str,
+    timeout: int = None,
```
Comment (Alex-Izquierdo) [inline src/eda_server/utils/subprocess.py:21]: This should be Optional if the default is None, isn't?

Comment (Alex-Izquierdo) [reply src/eda_server/utils/subprocess.py:21]: Should we have a sane default value for timeout?

**Review by Alex-Izquierdo — CHANGES_REQUESTED** [review]

**Review by Alex-Izquierdo — APPROVED** [review]
Verified

**Review by Alex-Izquierdo — APPROVED** [review]

**Review by ddonahue007 — APPROVED** [review]
lgtm

**Review by Alex-Izquierdo — APPROVED** [review]

**Pull request merged**